### PR TITLE
chore: add documentation for starknetid.ts

### LINKED ISSUE #1090: Improve JSDocs by adding examples of usages
Inspired by Viem docs, example:

https://viem.sh/docs/accounts/privateKey#usage

This is a big endeavor so it can be completed in multiple smaller PR-s (and more users can take it in parallel!)

IMPORTANT: add `@example` only to **exported** functions 

TODO utils:
- [ ] src/utils/calldata/validate 
- [ ] src/utils/calldata/formatter
- [ ] src/utils/calldata/propertyOrder
- [ ] src/utils/calldata/requestParser
- [ ] src/utils/calldata/responseParser
- [ ] src/utils/calldata/tuple
- [ ] src/utils/calldata/parser/index.ts
- [ ] src/utils/events/index.ts
- [ ] src/utils/hash/classHash.ts
- [ ] src/utils/hash/transactionHash/v3.ts

### PR DESCRIPTION
**Pull request by muheebyusufbaba1**
## Motivation and Resolution
Closes #1090

### RPC version (if applicable)

...

## Usage related changes

- Change 1.
- ...

## Development related changes

- Change 1.
- ...

## Checklist:

- [x] Performed a self-review of the code
- [x] Rebased to the last commit of the target branch (or merged it into my branch)
- [x] Linked the issues which this PR resolves
- [x] Documented the changes in code (API docs will be generated automatically)
- [x] Updated the tests
- [x] All tests are passing

### CODE REVIEW

File: src/utils/starknetId.ts [hunk 1]
```diff
@@ -24,6 +24,12 @@ function extractStars(str: string): [string, number] {
  * Decodes an array of BigInts into a string using the given algorithm.
  * @param {bigint[]} encoded - The encoded array of BigInts.
  * @return {string} The decoded string.
+ * @example
+ * ```typescript
+ * const result =  useDecoded([3015206943634620n])
```
Comment (ivpavici) [inline src/utils/starknetId.ts:29]:
```suggestion
 * const result = useDecoded([3015206943634620n])
```

**Comment by muheebyusufbaba1** [comment]
@ivpavici

File: src/utils/stark.ts [hunk 2]
```diff
@@ -28,6 +28,11 @@ import { isString } from './shortString';
  *
  * [Reference](https://github.com/starkware-libs/cairo-lang/blob/master/src/starkware/starknet/services/api/gateway/transaction.py#L54-L58)
  * @param jsonProgram Representing the compiled cairo program
+ * @example
```
Comment (ivpavici) [inline src/utils/stark.ts:31]: missing `@returns` here

File: src/utils/stark.ts [hunk 3]
```diff
@@ -48,6 +53,11 @@ export function decompressProgram(base64: CompressedProgram) {
 
 /**
  * Random Address based on random keyPair
+ * @example
```
Comment (ivpavici) [inline src/utils/stark.ts:56]: missing @returns here

File: src/utils/stark.ts [hunk 4]
```diff
@@ -58,6 +68,11 @@ export function randomAddress(): string {
  * Lowercase and hex prefix string
  *
  * @deprecated Not used internally, naming is confusing based on functionality
+ * @example
```
Comment (ivpavici) [inline src/utils/stark.ts:71]: we can remove for this one, it's deprecated

File: src/utils/stark.ts [hunk 5]
```diff
@@ -96,6 +111,11 @@ export function signatureToHexArray(sig?: Signature): ArraySignatureType {
 
 /**
  * Convert estimated fee to max fee with overhead
```
Comment (ivpavici) [inline src/utils/stark.ts:113]: missing `@param` and `@returns` here

File: src/utils/starknetId.ts [hunk 6]
```diff
@@ -24,6 +24,12 @@ function extractStars(str: string): [string, number] {
  * Decodes an array of BigInts into a string using the given algorithm.
  * @param {bigint[]} encoded - The encoded array of BigInts.
  * @return {string} The decoded string.
+ * @example
```
Comment (ivpavici) [inline src/utils/starknetId.ts:27]: @irisdv can you please confirm if these example's look ok?

File: src/utils/stark.ts [hunk 7]
```diff
@@ -96,6 +108,13 @@ export function signatureToHexArray(sig?: Signature): ArraySignatureType {
...
+ * @param {BigNumberish} estimatedFee - The estimated fee
+ * @param {number} [overhead = feeMarginPercentage.MAX_FEE] - The overhead added to the gas
+ * @returns {bigint} The maximum fee with overhead
```
Comment (ivpavici) [inline src/utils/stark.ts:113]: @example missning!

**Review by PhilippeR26 — CHANGES_REQUESTED** [review]
Nearly good, but as explained in Telegram https://t.me/c/2066195975/59 ,
please follow very rigorously the template of this PR https://github.com/starknet-io/starknet.js/pull/1100, especially for the examples : result always in a result variable, result displayed in a comment (//) (no assert).

**Comment by muheebyusufbaba1** [comment]
Reviews fixed @PhilippeR26

File: src/utils/stark.ts (hunk 7 above)
Comment (ivpavici) [reply src/utils/stark.ts:113]: @example still missing!

**Review by PhilippeR26 — CHANGES_REQUESTED** [review]
Is it normal to have no examples or missing @returns in some areas :
starks.ts : L79-106, L168-242 ?

**Pull request closed without merging**